drivers/gles2/rasterizer_scene_gles2.cpp

```diff
@@ -2054,7 +2055,11 @@ void RasterizerSceneGLES2::_render_render_list(RenderList::Element **p_elements,
 	if (p_env && !p_shadow && p_env->fog_enabled && (p_env->fog_depth_enabled || p_env->fog_height_enabled)) {
 		state.scene_shader.set_conditional(SceneShaderGLES2::FOG_DEPTH_ENABLED, p_env->fog_depth_enabled);
 		state.scene_shader.set_conditional(SceneShaderGLES2::FOG_HEIGHT_ENABLED, p_env->fog_height_enabled);
-		fog_max_distance = p_projection.get_z_far();
+		if (p_env->fog_depth_end > 0) {
```

Comment:
I wonder if this should use an epsilon (`CMP_EPSILON` if it's exposed here, or just `0.00001`) to take into account potential floating point precision issues?

Reply:
No, this is fine. The editor will always put zero in here so it is reliable to check it like this.